### DISCUSSION TITLE
How to read netcdf data from minio s3 server?

### DISCUSSION THREAD

**Discussion by forestbat** [pydata/xarray#8801] [Q&A] [answered]
This is my code:
```
storage_option = {'endpoint_url': 'url', 'key'='', 'secret'=''}
data_obj: xr.Dataset = xr.open_dataset(obj, chunks='auto', backend_kwargs={'remote_protocol':'s3', 'remote_options':storage_option})
```
And it crashed with:
`TypeError: NetCDF4BackendEntrypoint.open_dataset() got an unexpected keyword argument 'remote_protocol'`
My version of xarray is 2024.2.0.

**Comment by JessicaS11** [comment, marked as answer]
Hello @forestbat! I'm not familiar with minio, but it looks like the Xarray engine (which is defaulting to netcdf4 based on the file extenson of 'obj') cannot understand the access arguments you are passing in. If you are using a backend that integrates with minio, I would recommend supplying the `engine='yourbackend'` argument to `xr.open_dataset`. If not, [this discussion post](https://github.com/pydata/xarray/discussions/8626) discusses the use of fsspec to access different file system types and has some great explanations of what's going on behind the scenes.
